### DISCUSSION TITLE
dwgread generate json doc

### DISCUSSION THREAD

**Discussion by xiaoliangW** [LibreDWG/libredwg#900] [Q&A] [answered]
i use dwgread generate json file successfully, but i don't understand the  meaning of each word such as FILLMODE in HEADER, number in CLASSES, is  there any doc that explain these word?

**Comment by rurban** [comment, marked as answer]
Any DWG or DXF documentation would explain those terms. We are only using established terms by the standard 3rd party application vendors.

**Reply by xiaoliangW** [reply]
![20231225192435](https://github.com/LibreDWG/libredwg/assets/53174358/828d113d-c73c-4813-9af5-e4fcf74166ab)
two questions:
1. what dos these T、DT2、B  means? DXF 42 i think its DXF Group code 42
2.when i use dwgread dwg2geojson, it appear chinese unrecognizable characters, how to use it when it has chinese text.it dos not  have problem when i use dwgread dwg2json

**Comment by rurban** [comment]
All these types are explained in our docs, the oda spec and in the header.

As 2) do you have an example?

**Reply by xiaoliangW** [reply]
1.Could you provide some document addresses? i still don't find where to explain these json field
![json_field](https://github.com/LibreDWG/libredwg/assets/53174358/8cede922-490b-4f4d-ac86-6bbb6c959199)
2.i can't upload dwg in this place, i add some chinese words in test-data/2018/line.dwg.
![line-test](https://github.com/LibreDWG/libredwg/assets/53174358/b72ad6ea-a744-44bd-8b6f-8441efe81325)
to json,it is :
![line-json](https://github.com/LibreDWG/libredwg/assets/53174358/31d1d33f-82d9-4c0e-b8c2-2cc7db6c3463)
to geojson,it is:
![line-geojson](https://github.com/LibreDWG/libredwg/assets/53174358/426a824b-e1cf-49e8-befa-711a5416c7af)

**Comment by iisaxin** [comment]
您好，我刚接触libredwg,并且上周已经搭建好环境在VS2010，
简单的实体读取看了例程已经了解，
但是稍微复杂的就不会了，
例如下面的程序片断，
当读取一个线段实体时，他的颜色我会读取了，但是例如层和线型等是如何获得的呢？
for (i = 0; i < dwg.num_objects; i++)
{
switch (dwg.object[i].fixedtype)
{
case DWG_TYPE_LINE:
{line = dwg.object[i].tio.entity->tio.LINE;
color=dwg.object[i].tio.entity->color;
//layer=??
//ltype=??
多谢您的解答

**Comment by iisaxin** [comment]
你好呀，我上周才刚刚接触libredwg，
你的问题我暂时还没有接触到，
我的情况是刚刚把环境搭建好，用的vs2010，
我按照例程才刚刚可以读取到线，圆，文字，
现在我遇到一个问题，如果你可以解答那我是非常高兴的:)

例如下面的程序片断，
当读取一个线段实体时，他的颜色我会读取了，但是例如层和线型等是如何获得的呢？
for (i = 0; i < dwg.num_objects; i++)
{
switch (dwg.object[i].fixedtype)
{
case DWG_TYPE_LINE:
{line = dwg.object[i].tio.entity->tio.LINE;
color=dwg.object[i].tio.entity->color;//这个可以了
//layer=??//到这就不行了
//ltype=??//
多谢您的解答

其他内容还没有开始做



***@***.***
 
From: xiaoliangW
Date: 2023-12-26 16:08
To: LibreDWG/libredwg
CC: iisaxin; Comment
Subject: Re: [LibreDWG/libredwg] dwgread generate json doc (Discussion #900)
1.Could you provide some document addresses? i still don't find where to explain these json field
json_field.jpg (view on web)
2.i can't upload dwg in this place, i add some chinese words in test-data/2018/line.dwg.
line-test.jpg (view on web)
to json,it is :
line-json.jpg (view on web)
to geojson,it is:
line-geojson.jpg (view on web)

**Reply by xiaoliangW** [reply]
哥们加个联系方式聊聊？

**Reply by iisaxin** [reply]
对不起兄弟昨天忙没时间上来
我的电话13901398030
微信同号
急切希望沟通;)
